src/test/java/io/jenkins/plugins/coverage/model/CoveragePluginITest.java

```diff
@@ -0,0 +1,264 @@
+
+import java.io.IOException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import hudson.model.HealthReport;
+import hudson.model.HealthReportingAction;
+import hudson.model.Result;
+import hudson.model.Run;
+import static io.jenkins.plugins.coverage.model.Assertions.*;
+
+import io.jenkins.plugins.checks.api.ChecksAction;
+import io.jenkins.plugins.util.IntegrationTestWithJenkinsPerSuite;
+
+
+
+
+/**
+ * Integration test for the coverage API plugin.
+ *
+ * @author Thomas Willeit
+ *
+ */
+public class CoveragePluginITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private static final String JACOCO_FILE_NAME = "jacoco-analysis-model.xml";
+    private static final String JACOCO_LESS_FILE_NAME = "jacoco-codingstyle.xml";
+    private static final String COBERTURA_FILE_NAME = "cobertura.xml";
+    private static final String COBERTURA_FILE_NAME_2 = "cobertura2.xml";
+
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void CoveragePluginPipeline_0_JacocoInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles();
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult).isEqualTo(null);
+    }
+
+
+    @Test
+    public void CoveragePluginPipeline_1_JacocoInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_NAME);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('jacoco-analysis-model.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(6083, 6368-6083));
+    }
+
+    @Test
+    public void CoveragePluginPipeline_2_JacocoInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_NAME, JACOCO_LESS_FILE_NAME);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(12166, 12736-12166));
```

Comment:
It makes sense to use constants in the expressions for the whole test so this is better understandable. Something like `JACOCO_LARGE_MISSED_LINES`, `JACOCO_SMLL_COVERES_LINE`. Then you can let Java sum up these values for you.